synapse/util/caches/lrucache.py

```diff
@@ -58,6 +58,18 @@ def __init__(self, max_size, keylen=1, cache_type=dict):
 
         lock = threading.Lock()
 
+        def cache_len():
+            if size_callback is not None:
+                return sum(size_callback(node.value) for node in cache.itervalues())
```

Comment:
If you want to be awful you can move the if statement outside the function ``def`` and define the function twice...